c10/core/TensorImpl.h

```diff
@@ -637,6 +637,16 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     if (C10_UNLIKELY(
             sizes_strides_policy_ >=
             static_cast<uint8_t>(SizesStridesPolicy::CustomStrides))) {
+      if (is_python_dispatch()) {
```

Comment:
Prefer this to be out-of-line, either in the virtual method or a separate `is_contiguous_slow` out-of-line nonvirtual method